urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/examples/LCWTwoPlusOne.java

```diff
@@ -0,0 +1,348 @@
+                basis.add(q);
+            }
+        }
+        System.out.println();
+        c = new PermutationClass(basis);
+        s.clear();
+        for (RectangularizedPermutation qr : pr.children(9)) {
+            s.add(qr.getP());
+        }
+        for (Permutation q : new Permutations(10)) {
+            if (!s.contains(q) && c.containsPermutation(q)) {
+                System.out.println(q);
+                basis.add(q);
+            }
+        }
+        System.out.println();
+
+
+    }
+
+    private static class RectangularizedPermutation {
+
+        @Override
+        public int hashCode() {
+            int hash = 5;
+            hash = 97 * hash + (this.p != null ? this.p.hashCode() : 0);
+            hash = 97 * hash + (this.a != null ? this.a.hashCode() : 0);
+            hash = 97 * hash + (this.b != null ? this.b.hashCode() : 0);
+            return hash;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            final RectangularizedPermutation other = (RectangularizedPermutation) obj;
+            if (this.p != other.p && (this.p == null || !this.p.equals(other.p))) {
+                return false;
+            }
+            if (this.a != other.a && (this.a == null || !this.a.equals(other.a))) {
+                return false;
+            }
+            if (this.b != other.b && (this.b == null || !this.b.equals(other.b))) {
+                return false;
+            }
+            return true;
+        }
+        Permutation p;
+        Rectangle a;
+        Rectangle b;
+
+        public RectangularizedPermutation(Permutation p, Rectangle a, Rectangle b) {
+            this.p = p;
+            if (a == null) {
+                this.a = b;
+                this.b = null;
+            } else if (b == null) {
+                this.a = a;
+                this.b = null;
+            } else {
+                if (a.left < b.left) {
+                    this.a = a;
+                    this.b = b;
+                } else {
+                    this.a = b;
+                    this.b = a;
+                }
+            }
+        }
+
+        public Permutation getP() {
+            return p;
+        }
+
+        public HashSet<RectangularizedPermutation> children(int n) {
+            HashSet<RectangularizedPermutation> result = new HashSet<RectangularizedPermutation>();
+            result.add(this);
+            for (int i = 0; i < n; i++) {
+                HashSet<RectangularizedPermutation> r = new HashSet<RectangularizedPermutation>();
+                for (RectangularizedPermutation pr : result) {
+                    r.addAll(pr.children());
+                }
+                result = r;
+            }
+            return result;
+        }
+
+        public HashSet<RectangularizedPermutation> children() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 174-197 into a separate method.